lib/jsonrpccxx/client.hpp

```diff
@@ -68,12 +68,19 @@ class JsonRpcClient {
                               const json &params) {
     std::string id_str;
     json j = {{"method", name}};
-    if (std::get_if<int>(&id) != nullptr) {
-      j["id"] = std::get<int>(id);
-      id_str = std::to_string(std::get<int>(id));
-    } else {
-      j["id"] = std::get<std::string>(id);
-      id_str = std::get<std::string>(id);
+    if (const auto val = std::get_if<int>(&id)) 
+    {
+      j["id"] = *val;
+      id_str = std::to_string(*val);
```

Comment:
Is it make sense to create an extra `std::string id_str;`?
Why just not do:
`SPDLOG_DEBUG("Calling: {} method with id: {} and params: {} ", name, std::to_string(id), params);`

Reply:
it will not compile since there is no way to change variant to string just like that.
P.S this will for some reasons compile if you leave SPDLOG_DEBUG but if you change it to SPDLOG_ERROR it will fail